src/build/build_step.go

```diff
@@ -240,8 +278,10 @@ func buildTarget(tid int, state *core.BuildState, target *core.BuildTarget, runR
 				state.LogBuildResult(tid, target.Label, core.TargetCached, "Unchanged")
 			}
 			buildLinks(state, target)
+			filegroupsBuilt.Inc()
 			return nil
 		}
+		defer targetsBuilt.Inc()
```

Comment:
Below we do `retrieveArtifacts(...)` to check the cache but we'll still increment this counter. I'm not sure that's what we want?